lib/vapor/watch.ex

```diff
@@ -0,0 +1,33 @@
+defmodule Vapor.Watch do
+  use GenServer
+
+  alias Vapor.{
+    Provider
+  }
+
+  def start_link(state) do
+    GenServer.start_link(__MODULE__, state)
+  end
+
+  def init(state) do
+    schedule_refresh(state.plan)
```

Comment:
What are your thoughts on using [`:timer.send_interval/3`](https://erldocs.com/current/stdlib/timer.html?i=1&search=send_in#send_interval/3) instead of repeatedly calling `Process.send_after`?
Are we trying to avoid a backlog case where `handle_info(:refresh...` takes longer than the interval? and/or cleaning up the timer ref at process death?

Reply:
iirc `:timer.send_interval` has some performance problems. But regardless of that I used `Process.send_after` for exactly the reason you described. I didn't want to end up with a backlog of messages.